binarysearch.cpp

```diff
@@ -0,0 +1,36 @@
+
+#include <bits/stdc++.h>
+using namespace std;
+
+
+int binarySearch(int arr[], int l, int r, int x)
+{
+	if (r >= l) {
+		int mid = l + (r - l) / 2;
+
+		
+		if (arr[mid] == x)
+			return mid;
+
+		if (arr[mid] > x)
+			return binarySearch(arr, l, mid - 1, x);
+
+
+		return binarySearch(arr, mid + 1, r, x);
+	}
+
+
+	return -1;
+}
+
+int main(void)
+{
+	int arr[] = { 2, 3, 4, 10, 40 };
```

Comment:
Can you rewrite this part such that the user can give the array and the element to be found as input